sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/implementation/DocumentsImpl.java

```diff
@@ -401,21 +399,19 @@ public Mono<Response<SearchDocumentsResult>> searchGetWithResponseAsync(
         }
         UUID xMsClientRequestId = xMsClientRequestIdInternal;
         List<String> facetsConverted =
-                Optional.ofNullable(facets)
-                        .map(Collection::stream)
-                        .orElseGet(Stream::empty)
-                        .map((item) -> Objects.toString(item, ""))
-                        .collect(Collectors.toList());
+                (facets == null)
+                        ? new ArrayList<>()
+                        : facets.stream().map(item -> Objects.toString(item, "")).collect(Collectors.toList());
         String highlightFieldsConverted =
                 JacksonAdapter.createDefaultSerializerAdapter().serializeList(highlightFields, CollectionFormat.CSV);
         String orderByConverted =
                 JacksonAdapter.createDefaultSerializerAdapter().serializeList(orderBy, CollectionFormat.CSV);
         List<String> scoringParametersConverted =
-                Optional.ofNullable(scoringParameters)
-                        .map(Collection::stream)
-                        .orElseGet(Stream::empty)
-                        .map((item) -> Objects.toString(item, ""))
-                        .collect(Collectors.toList());
+                (scoringParameters == null)
+                        ? new ArrayList<>()
```

Comment:
would `Collections.emptyList()` work here (warning, it's immutable). otherwise should this be `ArrayList(0)` if there's low chance of anybody adding anything to this ?

Reply:
good to know.